packages/core/workflows-sdk/src/utils/_playground.ts

```diff
@@ -53,6 +55,28 @@ workflow2()
     console.log(res.result)
   })
 
+type MyWorkflowHooks = {
+  beforeCreate: (id: string) => string
+}
+
+const workflowWithHooks = createWorkflowWithHooks(
+  (useHook: UseHook<MyWorkflowHooks>) => {
```

Comment:
hint: This allows us to wrap the hook util under the hook and it fully typed, the first argument being the hook name coming from the type define above `MyWorkflowHooks ` and the argument is also typed based on the above type `MyWorkflowHooks`